cpp/include/cudf/detail/utilities/batched_memset.hpp

```diff
@@ -50,9 +50,14 @@ void batched_memset(std::vector<cudf::device_span<T>> const& bufs,
   // define task and bytes parameters
   auto const num_bufs = bufs.size();
 
+  auto host_pinned_bufs = cudf::detail::make_pinned_vector_async<cudf::device_span<T>>(
```

Comment:
We should maybe make the first argument a template so to make it a responsibility of the caller to pass a pageable or pinned vector as needed. Thoughts @vuule.

Also, perhaps we need the same changes for `batched_memcpy` utility as well.

Reply:
I was just looking at this now. AFAICT this function should take a host_span and the caller should use host_vector instead of std::vector
now sure how we missed this when we added the API.

Reply:
Agreed.